src/Stratis.Bitcoin.Features.Consensus/CoinViews/DBreezeCoinView.cs

```diff
@@ -213,15 +215,25 @@ public Task SaveChangesAsync(IEnumerable<UnspentOutputs> unspentOutputs, IEnumer
 
                         this.SetBlockHash(transaction, nextBlockHash);
 
-                        all.Sort(UnspentOutputsComparer.Instance);
-                        foreach (UnspentOutputs coin in all)
+                        var toRemove = all.Where(utxo => utxo.IsPrunable == true).ToList();
+                        toRemove.Sort(UnspentOutputsComparer.Instance);
```

Comment:
why not instead of creating a list enumerate all.where(...).OrderBy() ?
same for code below

Reply:
look my comment above https://github.com/stratisproject/StratisBitcoinFullNode/pull/2484#discussion_r225306605